### PR TITLE
ISAICP-5411: Discussion SEO metatags.

### PR DESCRIPTION
**Pull request by idimopoulos**

### CODE REVIEW

File: web/profiles/joinup/joinup.behat.inc [hunk 1]
```diff
@@ -721,7 +740,6 @@ class JoinupSubContext extends DrupalSubContextBase {
         'headline' => 'field_news_headline',
         'keywords' => 'field_keywords',
         'logo' => 'field_news_logo',
-        'publication date' => 'published_at',
```
[the remark this answers is not in the document]
Comment (idimopoulos) [reply web/profiles/joinup/joinup.behat.inc]: It is a duplicated entry :) There is another one below.

File: config/sync/metatag.metatag_defaults.node__discussion.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+uuid: c440cbac-8226-4beb-8231-07b5c3eea705
+langcode: en
+status: true
+dependencies: {  }
+id: node__discussion
+label: 'Content: Discussion'
+tags:
+  schema_social_media_posting_name: '[node:title]'
+  schema_social_media_posting_about: '[node:field_keywords]'
+  schema_social_media_posting_author: 'a:4:{s:5:"@type";s:6:"Person";s:3:"@id";s:17:"[node:author:url]";s:4:"name";s:26:"[node:author:display-name]";s:3:"url";s:17:"[node:author:url]";}'
```
[the remark this answers is not in the document]
Comment (idimopoulos) [reply config/sync/metatag.metatag_defaults.node__discussion.yml]: I guess for faster porting from D7?? Do you want me to open a ticket for that?

File: web/modules/custom/joinup_seo/joinup_seo.module [hunk 3]
```diff
@@ -21,3 +21,44 @@ function joinup_seo_simple_sitemap_sitemap_types_alter(array &$sitemap_types): v
...
+ * When a field contains multiple values, the normal way of adding one entry for
+ * each delta is to use the Pivot. Pivot will allow the field to be rendered as
+ * an array value objects instead of a single object.
+ * In order to render all deltas in the field, the tag tokens must include all
+ * deltas separated by comma, e.g.
+ * '[node:field_file:0:entity:url],[node:field_file:1:entity:url]' etc.
+ * When the field has unlimited cardinality though, it is impossible to do such
+ * a thing. Thus, dynamically change the metatags for the entity to include a
+ * specific number of tokens with delta that ensure that all values are
+ * rendered.
```
[the remark this answers is not in the document]
Comment (idimopoulos) [reply web/modules/custom/joinup_seo/joinup_seo.module]: Possibly.. Not definitively however.. The issue with this is that I do not have any significant data to make any assumptions of how it should work. I can open a ticket to present my solution and ask if this is the correct approach.. The problem however is the complexity of different cases in the schema tags.
For example, Pivot, is mainly meant to deal with sub objects. E.g. we have a document which has a schema description, but it references a list of files which has its own schema description. The Pivot would take care of converting the references into an array of objects.

However, when it comes to this specific case, I think to generalize this solution would be wrong. The most proper approach would be to be able to define the schema for the file entity itself and then simply provide the field token and have it calculated automatically. Having all entities be checked for whether they have a token, identify the token and alter it, would be very complex. Consider cases where the token does not come on its own but as part of a text. We would have to either duplicate the tests or duplicate the tokens in the text.
Other than that, there can be cases where the field is a link or a text field. 

Again, I did not have enough information to assume that this is a generic approach or a specific solution as the tests are not that helpful.

File: config/sync/metatag.metatag_defaults.node__discussion.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (claudiu-cristea) [reply config/sync/metatag.metatag_defaults.node__discussion.yml]: No, it's OK.

**Pull request merged**